docs/tables.md

```diff
@@ -186,4 +186,58 @@ Reduced Scans
+
+| Error Stage | Explanation |
+| ----------- | ----------- |
+| setup       | The initial setup phase for the test (mustering resources, opening ports, etc.) |
+| dial        | The initial TCP dial connection to the remote |
+| tls         | The TLS handshake. `HTTPS` only. The [SNI header](https://en.wikipedia.org/wiki/Server_Name_Indication) containing the test domain is sent in this stage |
+| write       | Writing to the remote. For non-`HTTPS` tests this is where the domain is sent. |
+| read        | Reading from the remote |
+| http        | Verification of HTTP headers. `HTTP/S` only |
+| content     | Verification that the returned content matches the expected content. These are the most common types of errors and represent things like blockpages. |
+| final       | Reaching the final stage without encountering any problems in the previous stages. |
+| unknown     | The stage of the error is not known. Usually these are new errors which should be investigated and classified  |
+
+##### Error Class
+
+Protocol errors are mostly defined by the Network Error Logging standard's [Predefined Network Error Types](https://www.w3.org/TR/network-error-logging/#predefined-network-error-types). We add one extended error type, `tcp.no_route`. These errors may represent normal network failures and noise, but they may also expose packets inserted by a middlebox to disrupt a connection. For example China and Iran are both known to use [TCP Reset Attacks](https://en.wikipedia.org/wiki/TCP_reset_attack) as a censorship method.
+
+Mismatch Errors are used when the connection is successful, but the content received does not match the content expected. This can happen in the case of blockpages, remote servers with unusual behavior, or complicated CDN networks serving many sites.
+
+| Error Class             | Explanation |
+| ----------------------- | ----------- |
+| success                 | The test completed successfully and no interference was detected |
+| invalid                 | The test was invalid, usually a test system failure |
+| unknown                 | The class of the error is not known. Usually these are new errors which should be investigated and classified |
+|                         | |
+| **Protocol Errors**     | [Predefined Network Error Types](https://www.w3.org/TR/network-error-logging/#predefined-network-error-types) |
+|                         | |
+| tcp.address_unreachable | The IP address is unreachable |
+| tcp.no_route            | No route to the IP address could be found |
+| tcp.refused             | The TCP connection was refused by the server |
+| tcp.closed              | The TCP connection was closed by the server. Could indicate an inserted `FIN` packet |
```

Comment:
Do we ever see this? I don't think this is a valid error. Any TCP connection can close. Perhaps we should have a http/empty_response or content/empty instead.

Reply:
We often see this, and it doesn't always indicate that the response was empty. (Sometimes there was a partial response, then the error.)

I haven't looked to much into individual cases, but the fact that 60% of these errors come from China (vs only 1-2% of the general data) makes me think there's something suggestive about these errors indicating network interference and not just empty responses from servers.

![C8YBhHSNXwnqw9L](https://user-images.githubusercontent.com/1127209/110182768-6de58400-7ddb-11eb-9698-63b4a8380a18.png)